finesse/hardware/data_file_writer.py

```diff
@@ -65,11 +64,7 @@ def _get_stepper_motor_angle() -> tuple[float, bool]:
         else:
             return (float("nan"), True)
     except Exception as error:
-        pub.sendMessage(
-            f"device.error.{config.STEPPER_MOTOR_TOPIC}",
-            instance=DeviceInstanceRef(config.STEPPER_MOTOR_TOPIC),
-            error=error,
-        )
+        stepper.send_error_message(error)
```

Comment:
It feels a bit odd to handle _externally_ an error generated by doing something with a device by calling a method of the device itself. Would it make more sense to check that things are ok in the methods being called and, if not, call the `send_error_message` _internally_?

This applies to other devices using this same approach.

Reply:
I know what you mean -- it is a bit odd. It doesn't feel entirely right to catch the error inside the method either though, because we want to pass the error on to the caller. (The errors will be caught and retransmitted as a `pubsub` message if this value is requested via a `pubsub` message though.)

I feel in general this file needs another look at because I'm not convinced about the design. I tried rewriting it to use `pubsub` messages instead of importing these classes directly though and it ended up being up being horrible :cry:.